### ISSUE TITLE
No way to override style of a floating <OverflowMenu>in v9

### ISSUE THREAD

**Issue by s100** [carbon-design-system/carbon#2876]
### Summary

I would open this issue against [https://github.com/carbon-design-system/carbon-components-react] but apparently this is the new place...

In Carbon V9 (specifically, `carbon-components-react@6`) it is not possible to override the styling of an `<OverflowMenu floatingMenu />` to *e.g.* make it wider. Specifically, the options themselves. As with nearly all Carbon components it is possible to pass a `className` prop into the component, which can then be targeted to alter rendering. Unfortunately, when the menu options are floating, they appear at a completely different location in the DOM from where the injected `className` goes. So all we can do is alter the rendering of the menu icon.

### Justification

This would be useful for making the floating menu bigger, so that longer menu options are easier to read.

### Desired UX and success metrics

Something like: when a `<OverflowMenu floatingMenu optionsClassName='x' />` is opened, the `<ul>` which gets appended to the DOM should have `class='x'` on it.

### "Must have" functionality

That's about it!

### Specific timeline issues / requests

Soon as you like, really. This is a simple request so if this isn't likely to appear in the next week I may be able to put together a PR myself.

### Available extra resources

I'll try to patch this myself if I have to...

**Comment by asudoh** [comment]
Hi 👋 thank you for reporting - While your `optionsClassName` may make sense, one thing that may be worth mentioning is that overflow menu looks through ancestor nodes of the trigger button in DOM for one with `data-floating-menu-container` attribute and puts its menu body in there. `<body>` is the last resort. That said, you can add CSS class to that container and create override style for menu body with it. Hope this helps!

**Comment by s100** [comment]
@asudoh That may actually be useful, thank you!

**Comment by s100** [comment]
@asudoh Actually this proved more troublesome than I was expecting because something in `<OverflowMenu>`'s layout calculations is computing offsets (`top` and `left`) for the absolutely positioned menu body under the assumption that it is being absolutely positioned relative to `<body>`. In fact the menu body is [absolutely positioned](https://developer.mozilla.org/en-US/docs/Web/CSS/position) relative to its [nearest positioned ancestor](https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block), which defaults to `<body>` if there are no positioned ancestors, but in my case is a different element. So, if the NPA is at (200px, 400px), the menu itself renders at roughly (200px, 400px) relative to that element, or (400px, 800px).

I have not yet figured out what the specific problem here is, the code is complex. Currently I am working around this by passing in a replacement `getMenuOffset` prop which carries out the right calculations.

**Comment by asudoh** [comment]
> Currently I am working around this by passing in a replacement getMenuOffset prop which carries out the right calculations.

It's the right way to go. BTW if you have some demo code that shows the layout challenge you have, we may consider taking the non-`<body>` floating menu container's bounding rect into account.

**Comment by s100** [comment]
@asudoh Small demo of the issue I was seeing:

```html
<!DOCTYPE html>
<html>
  <head>
    <title></title>
    <link rel="stylesheet" href="https://unpkg.com/carbon-components@9.91.3/css/carbon-components.css"></link>
    <script crossorigin src="https://unpkg.com/react@16/umd/react.development.js"></script>
    <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"></script>
    <script crossorigin src="https://unpkg.com/prop-types@15.7.2/prop-types.js"></script>
    <script crossorigin src="https://unpkg.com/carbon-icons@7.0.7/dist/carbon-icons.js"></script>
    <script crossorigin src="https://unpkg.com/classnames@2.2.6/index.js"></script>
    <script crossorigin src="https://unpkg.com/carbon-components-react@6.115.3/umd/carbon-components-react.js"></script>
    <script crossorigin src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
    <script type="text/jsx">
      const { OverflowMenu, OverflowMenuItem } = CarbonComponentsReact

      window.addEventListener('load', () => {
        ReactDOM.render(<div
          data-floating-menu-container
          style={{
            width: '500px',
            height: '500px',
            backgroundColor: 'yellow',
            position: 'relative',
            left: '100px',
            top: '100px'
          }}
        >
          <OverflowMenu className='overflow-menu' floatingMenu>
            <OverflowMenuItem itemText='Menu item 1' />
            <OverflowMenuItem itemText='Menu item 2' />
          </OverflowMenu>
        </div>, document.querySelector('.root'))
      })
    </script>
  </head>
  <body>
    <div class="root"></div>
  </body>
</html>
```

![image](https://user-images.githubusercontent.com/9932290/58700038-eca3c680-8396-11e9-911a-e88d1d7bffc3.png)

Note how the vector from the top left corner of the screen to the bottom midpoint of the menu icon is the same as the vector from the bottom midpoint of the menu icon to the tip of the arrow in the menu body. This indicates that the menu body is being positioned relative to its nearest positioned ancestor (the yellow `<div>`) but the *calculations* for its position assume that it is being positioned relative to `<body>`.

**Comment by s100** [comment]
Should I raise this relative positioning bug as a separate issue?

**Comment by asudoh** [comment]
@s100 Yes it will be great - Thanks!

**Comment by s100** [comment]
@asudoh: raised #2907.

**Comment by s100** [comment]
@asudoh: Do you know when a new version of `carbon-components-react@6` containing #2389 will be available?

**Comment by asudoh** [comment]
@s100 #2389 seems a different issue - Any other issue you are referring to?

**Comment by s100** [comment]
@asudoh Ah, sorry, I was referring to https://github.com/carbon-design-system/carbon-components-react/pull/2389, which closes this issue (#2876)

**Comment by s100** [comment]
@asudoh Do you know when a new version of carbon-components-react@6 containing https://github.com/carbon-design-system/carbon-components-react/pull/2389 will be available?
